### PR TITLE
Add partition and primary key properties

### PR DESCRIPTION
**Pull request by calumabarnett**
This PR will add the following properties to the table metadata schema at the table level:

- `partitions`
- `primary_key`

### CODE REVIEW

File: table/v1.3.0.json [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+                        "type": "string",
+                        "title": "The column name. Should be lower case with underscores, not spaces",
+                        "pattern": "[a-z0-9_]+",
+                        "examples": [
+                            "employee_number"
+                        ]
+                    },
+                    "type": {
+                        "type": "string",
+                        "title": "The data type. We use a limited set of data types for cross compatibility between Spark, R, Pandas etc. See lookup here: https://github.com/moj-analytical-services/dataengineeringutils/blob/master/dataengineeringutils/data/data_type_conversion.csv",
```
Comment (MrAlecJohnson) [inline table/v1.3.0.json:34]: I don't think this link is up to date - the repo is archived and the list of types doesn't include the new decimal.

Maybe out of scope for this pull request, but could this table belong somewhere else instead? Maybe the readme for metadata_schema?

File: table/v1.3.0.json [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+            "type": "string",
+            "title": "A description of what this table contains"
+        },
+        "name": {
+            "type": "string",
+            "title": "The name of the table in the database"
+        },
+        "data_format": {
+            "type": "string",
+            "title": "The format of the data in s3, and instruction on how to parse, see here https://github.com/moj-analytical-services/dataengineeringutils/blob/ae295caf93c75c80510abf0c74865939c94d3e70/dataengineeringutils/glue.py#L45",
```
Comment (MrAlecJohnson) [inline table/v1.3.0.json:105]: As that repo is archived, is this bit of etl_manager now the equivalent: https://github.com/moj-analytical-services/etl_manager/blob/912dc34630e6903a0b55f4d3238a521d3cef2fb4/etl_manager/meta.py#L30

**Review by MrAlecJohnson — APPROVED** [review]
Changes look good to me 👍 

I'm wondering if it's worth changing the outdated links to dataengineeringutils. Though I realise they aren't  strictly part of this pull request

**Comment by calumabarnett** [comment]
@MrAlecJohnson I'll get this merged in and will do another quick PR to address your comments.

**Pull request merged**